android/automation.md

```diff
@@ -0,0 +1,175 @@
+# Automation and continuous integration systems
+
+Our Android projects share a similar automation model:
+- Taskcluster is our primary CI: we use it for as much automation as
+possible, including our core builds and releases, because it's a
+flexible first party service we can rely on.
+- Travis is secondary: we use it to get automation working quickly or get
+support for tools that won't work on Taskcluster.
+
+## Taskcluster
+
+[Taskcluster][tc] is a generic task execution framework developed and used by Mozilla to automate continuous integration and release processes.
+
+We currently use Taskcluster to:
+* Build, test and run code tools for all master commits and pull requests
+(including contributor PRs). See [Code quality tools](#code-quality-tools)
+for additional information.
+* Build (unsigned) release builds whenever a release is created on GitHub
+* Automatically import new translations from the L10N repository.
+
+To see a Taskcluster configuration for your project, see
+`<project-root>/.taskcluster.yml` (here's an
+[example in focus-android][tc yml]).
+
+Taskcluster runs using docker: see our [docker guide](docker_guide.md) for
+more information.
+
+## Travis
+
+[Travis] is an easy-to-configure continuous integration environment with
+great support for GitHub.
+
+We currently use Travis to:
+* Build, test and run code tools for all master commits and pull requests (including contributor PRs).
+* Execute sputnik for posting automated code feedback in pull requests
```

Comment:
I think sputnik is less helpful now that we are mostly writing Kotlin code and not Java anymore. Sputnik runs detekt and ktlint too, but in outdated version and they are not using the configuration from the repo like they do for findbugs etc.
With that I wouldn't add sputnik to a new repository anymore. And because of false-positives I wonder whether we should remove it from existing projects too. The only nice thing about it are inline comments on PRs. But we can live without that too.

Reply:
I've noticed that too: I'll remove any notes about sputnik.

I think if we encourage devs to set up pre-push hooks, it's much less useful anyway.